client/lib/media/utils.js

```diff
@@ -108,7 +108,8 @@ const MediaUtils = {
 		} else if ( media.extension ) {
 			extension = media.extension;
 		} else {
-			extension = path.extname( url.parse( media.URL || media.file || media.guid || '' ).pathname ).slice( 1 );
+			const pathname = url.parse( media.URL || media.file || media.guid || '' ).pathname || '';
```

Comment:
pathname was sometimes null, so it would throw an error in `path.extname( pathname )` see: https://circleci.com/gh/Automattic/wp-calypso/16314